dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -87,6 +94,18 @@ def save_partner
       self.regional_partner_id = sanitize_form_data_hash[:regional_partner_id]
     end
 
+    before_save :enroll_user, if: -> {properties_changed?}
+    def enroll_user
+      return unless pd_workshop_id
+
+      ::Pd::Enrollment.find_or_create(
```

Comment:
It's [find_or_create_by](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by) (missing `_by` at the end), and we might as well use the bang version `find_or_create_by!`

Note there is also a [first_or_create](https://apidock.com/rails/ActiveRecord/Relation/first_or_create), but it works on a relation (where clause) and I find it to be less readable.